gcloud-java-contrib/gcloud-java-nio/src/test/java/com/google/gcloud/storage/contrib/nio/ITGcsNio.java

```diff
@@ -0,0 +1,351 @@
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * Integration test for gcloud-nio. This test actually talks to GCS (you need an account).
+ * Tests both reading and writing.
+ *
+ * You *must* set the GOOGLE_APPLICATION_CREDENTIALS environment variable
+ * for this test to work. It must contain the name of a local file that contains
+ * your Service Account JSON Key.
+ *
+ * The instructions for how to get the Service Account JSON Key are
+ * at https://cloud.google.com/storage/docs/authentication?hl=en#service_accounts
+ *
+ * The short version is this: go to cloud.google.com/console,
+ * select your project, search for "API manager", click "Credentials",
+ * click "create credentials/service account key", new service account,
+ * JSON. The contents of the file that's sent to your browsers is your
+ * "Service Account JSON Key".
+ *
+ */
+@RunWith(JUnit4.class)
+public class ITGcsNio {
+
+  private static final List<String> FILE_CONTENTS = ImmutableList.of(
+      "Tous les êtres humains naissent libres et égaux en dignité et en droits.",
+      "Ils sont doués de raison et de conscience et doivent agir ",
+      "les uns envers les autres dans un esprit de fraternité.");
+
+  private static final Logger log = Logger.getLogger(ITGcsNio.class.getName());
+  private static final String BUCKET = RemoteGcsHelper.generateBucketName();
+  private static final String SML_FILE = "tmp-test-small-file.txt";
+  private static final int SML_SIZE = 100;
+  // it's big, relatively speaking.
+  private static final String BIG_FILE = "tmp-test-big-file.txt";
+  // arbitrary size that's not too round.
+  private static final int BIG_SIZE = 2 * 1024 * 1024 - 50;
+  private static final String PREFIX = "tmp-test-file";
+  private static Storage storage;
+  private static StorageOptions storageOptions;
+
+  private final Random rnd = new Random();
+
+  @BeforeClass
+  public static void beforeClass() throws IOException {
+    // loads the credentials from local disk as par README
+    RemoteGcsHelper gcsHelper = RemoteGcsHelper.create();
+    storageOptions = gcsHelper.options();
+    storage = storageOptions.service();
+    // create and populate test bucket
+    storage.create(BucketInfo.of(BUCKET));
+    fillFile(storage, SML_FILE, SML_SIZE);
+    fillFile(storage, BIG_FILE, BIG_SIZE);
+  }
+
+  @AfterClass
+  public static void afterClass() throws ExecutionException, InterruptedException {
+    if (storage != null && !RemoteGcsHelper.forceDelete(storage, BUCKET, 5, TimeUnit.SECONDS)) {
+      if (log.isLoggable(Level.WARNING)) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/mziccard/gcloud-java/file/2295976424/issues/source?bid=3123107&fileBranchId=3162887#l96)